tsl/src/compression/api.c

```diff
@@ -823,16 +918,18 @@ tsl_decompress_chunk(PG_FUNCTION_ARGS)
 	if (!ht->fd.compressed_hypertable_id)
 		ereport(ERROR, (errcode(ERRCODE_INTERNAL_ERROR), errmsg("missing compressed hypertable")));
 
-	if (!ts_chunk_is_compressed(uncompressed_chunk))
+	if (ts_is_hypercore_am(uncompressed_chunk->amoid))
+		set_access_method(uncompressed_chunk_id, "heap");
```

Comment:
This change looks like running decompress_chunk on hypercore TAM won't actually decompress the chunk but rather only switch its access method to heap.

Is that the intended usage? Run decompress_chunk twice in order to decompress a chunk?

Reply:
Setting an access method rewrites the whole table. This is something that PG does automatically when calling ALTER TABLE  ... SET ACCESS METHOD. So, the table will be decompressed after this command.

The only thing we are doing here is to make `decompress_chunk()` basically be equivalent to `ALTER TABLE ... SET ACCESS METHOD heap`.